plugin/segment/segment.go

```diff
@@ -0,0 +1,123 @@
+package segment
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/store"
+	"github.com/google/uuid"
+	"github.com/segmentio/analytics-go"
+
+	"go.uber.org/zap"
+)
+
+// EventType is the segment track event name.
+type EventType string
+
+var (
+	// InstanceEventType is the track event for instance.
+	InstanceEventType EventType = "instance"
+)
+
+// Segment is the metrics collector https://segment.com/.
+type segment struct {
+	l            *zap.Logger
+	identify     string
+	client       analytics.Client
+	store        *store.Store
+	reporterList []Reporter
+}
+
+// Workspace is the instance for console application.
+type Workspace struct {
+	License string
+}
+
+// Service is the service for Segment.
+type Service interface {
+	Close()
+	Report(ctx context.Context)
+	Track(event EventType, properties analytics.Properties)
+	Identify(workspace *Workspace)
+}
+
+// NewService creates a new instance of Segment
+func NewService(l *zap.Logger, key string, store *store.Store) (Service, error) {
+	identify, err := initIdentify(store)
+	if err != nil {
+		return nil, err
+	}
+
+	client := analytics.New(key)
+
+	return &segment{
+		l:        l,
+		identify: identify,
+		client:   client,
+		store:    store,
+		reporterList: []Reporter{
+			&InstanceReporter{},
+		},
+	}, nil
+}
+
+func initIdentify(store *store.Store) (string, error) {
+	ctx := context.Background()
+	configCreate := &api.SettingCreate{
```

Comment:
this probably should be in the server.go and pass the id here.

Otherwise, we have a dependency reversal since this is a Bytebase server server-specific, and the plugin should have no knowledge of it.